compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/CodeGenerator.java

```diff
@@ -83,10 +85,15 @@ private CompiledJarFile generate(BPackageSymbol packageSymbol) {
         populateExternalMap(jvmPackageGen);
 
         //Rewrite identifier names with encoding special characters
-        JvmDesugarPhase.encodeModuleIdentifiers(packageSymbol.bir, Names.getInstance(this.compilerContext));
+        HashMap<String, String> originalIdentifierMap = JvmDesugarPhase
+                .encodeModuleIdentifiers(packageSymbol.bir, Names.getInstance(this.compilerContext));
 
         // TODO Get-rid of the following assignment
         packageSymbol.compiledJarFile = jvmPackageGen.generate(packageSymbol.bir, true);
+
+        //Revert encoding identifier names
+        JvmDesugarPhase.replaceEncodedModuleIdentifiers(packageSymbol.bir, Names.getInstance(this.compilerContext),
```

Comment:
There are two ways to fix this issue.

1) Decode all encoded identifier names after codegen.
2) Track the original identifier names before encoding and then replace encoded identifier names after codegen with original strings.

1 option didn't work since for some string original string and decoded string retrieved via encoded string was not the same.
ex.If you encode "Will.be&encoded", then decode it you will get "Will\.be\&encoded"

Reply:
@MaryamZi @HindujaB

Reply:
@warunalakshitha, but doesn't that mean that our decoding logic is not sound? Couldn't this be an issue when decoding at runtime?

Reply:
This is because we have 3 different representations for an identifier for different purposes. 
1. identifiers that user defined before encoding (with escape characters) - Eg. `'Will\.be\&encoded`
2. identifiers that the user intended to have (without escaped characters) - Eg. `Will.be&encoded` 
     - Used for member access, etc. 
3. identifiers that are encoded (for jvm back-end) - Eg. `Will$0046be&encoded` 

The decode function will always return the identifier to its original state (with escaped characters)
Therefore, +1 to the approach.